Update dashboard to design v122 (Part 1)

somebody1234 opened the pull request:
### Pull Request Description
- Close #9886
  - Update style of buttons in Drive Bar
  - Move "Home" page into a "Start" dialog
  - Remove icons that are no longer needed
  - Remove Backend Switcher in top bar - they have now been converted to categories

- Incidental changes
  - Refactor Backend provider so that both Remote and Local backends are available.
    - This was done because both Cloud and Local backends are now easily accessible by switching tabs - the Local backend no longer has its own views with the hidden category switcher

### Important Notes
None

### Code review tips
- There are a lot of changes in Settings components, for example (and in general, any component that needs to use a backend). However, most of the changes are identical, so only the approach needs to be reviewed - although the settings pages and the other pages should probably be reviewed separately.

### Testing instructions
- Make sure new/changed features work, as mentioned above (drive bar, start page, drag-n-drop)
- Because of the backend changes, all flows in general should be checked, at least at a basic level:
  - Sign up/sign in
  - Offline mode (bottom link on login screen on desktop app)
  - Checking that all settings pages work, both when logged in and not logged in

### Screencasts

https://github.com/enso-org/enso/assets/4046547/d8f385e2-3355-404f-a346-035917d40cf9

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] ~~The documentation has been updated, if necessary.~~
- [x] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [x] ~~Unit tests have been written where possible.~~

somebody1234 committed:
Refactor start page into a dialog; update Drive Bar styles

somebody1234 committed:
Remove obsolete `onSpinnerStateChange` event

somebody1234 committed:
Add new UI for dropzone

somebody1234 committed:
Convert Cloud and Local from backend types to categories

Lots more refactors still need to be done so that both backends are always present

somebody1234 committed:
WIP: Split useBackend into useRemoteBackend and useLocalBackend

somebody1234 committed:
Fix type errors

somebody1234 committed:
Remove `useRemoteBackendStrict` from Settings pages

somebody1234 committed:
Minor UI fixes and improvements

somebody1234 committed:
Fixes for offline mode

somebody1234 committed:
Add variant to `UnstyledButton` for new button style

somebody1234 commented:
I guess I'll avoid stacking the move to tabs on top of this PR, as this is a big PR so there may be a lot of changes needed.

somebody1234 committed:
Prettier

somebody1234 committed:
Merge branch 'develop' into wip/sb/dashboard-122

somebody1234 committed:
Fix type errors

somebody1234 committed:
Fix lints

somebody1234 committed:
Merge branch 'develop' into wip/sb/dashboard-122

somebody1234 committed:
Prettier

somebody1234 committed:
Fix E2E tests

somebody1234 committed:
Fix ESLint config

somebody1234 committed:
Lighten extra columns for assets table

somebody1234 committed:
Adjust ESLint ignores

somebody1234 committed:
Merge branch 'develop' into wip/sb/dashboard-122

somebody1234 committed:
Merge branch 'develop' into wip/sb/dashboard-122

somebody1234 committed:
Fix lint errors

somebody1234 committed:
Fix error when `build.json` is missing

somebody1234 committed:
Fix import error

somebody1234 committed:
Merge branch 'develop' into wip/sb/dashboard-122

somebody1234 committed:
Merge branch 'develop' into wip/sb/dashboard-122

somebody1234 committed:
Prettier

PabloBuchu commented:
1. This loader is barely moving when running cloud project

<img width="139" alt="Screenshot 2024-05-20 at 16 16 55" src="https://github.com/enso-org/enso/assets/2855109/50e94fd9-1363-4726-a448-d04712d73fb4">

2. After cmd+r when having project open in progress I no longer see calls `projects/<project_id>` so I had to reload again to gain access to opened project

somebody1234 commented:
> This loader is barely moving when running cloud project

weird, i seem to be able to reproduce on develop. seems like it might be a regression

somebody1234 committed:
Fix broken spinner animations

somebody1234 commented:
> After cmd+r when having project open in progress I no longer see calls `projects/<project_id>`

ah. i guess we should store the "currently open project" *immediately* when the user clicks play, rather than after it actually opens. it's late though so probably won't be fixed today (and possibly will fix on the tabs PR instead.)

somebody1234 committed:
Fix projects not continuing to load on reload

MrFlashAccount commented:
@somebody1234 could you please rebase onto the latest changes so I can review the PR?

somebody1234 committed:
Merge branch 'develop' into wip/sb/dashboard-122

somebody1234 commented:
@MrFlashAccount latest changes pulled.

MrFlashAccount commented on app/ide-desktop/lib/dashboard/src/components/AriaComponents/Dialog/Dialog.tsx:
```diff
@@ -8,6 +8,7 @@ import * as twv from 'tailwind-variants'
 import * as aria from '#/components/aria'
 import * as ariaComponents from '#/components/AriaComponents'
 import * as portal from '#/components/Portal'
+import WindowButton from '#/components/styled/WindowButton'
```
We have CloseButton, You don't have to use `WindowButton` or something

MrFlashAccount commented on app/ide-desktop/lib/dashboard/src/components/AriaComponents/Dialog/Dialog.tsx:
```diff
@@ -150,6 +152,11 @@ export function Dialog(props: types.DialogProps) {
                     {typeof children === 'function' ? children(opts) : children}
                   </div>
+                  {closeButton === 'floating' && (
+                    <div className="m-floating-buttons absolute flex gap-1">
```
Don't use `m-floating-buttons` or any other custom classes, because they make no sense and breaks style deduplication

somebody1234 replied:
style deduplication should not be a concern though

MrFlashAccount replied:
It's always a concern

somebody1234 replied:
but why?

MrFlashAccount replied:
Because:

1. `!important` hell
2. m-floating-buttons - how'd I understand how much it is in pixels?
3. tw-merge doesn't know `m-floating-buttons` 
4. We use the atomic approach, and style deduplication is one of the most important features.
5. We have tailwind-variants - if you want to add styles like 'button' use this lib instead.

somebody1234 replied:
> `!important` hell

should not happen, an `m-` is still an `m-`.

> how'd I understand how much it is in pixels?

you're not supposed to

> tw-merge doesn't know `m-floating-buttons`

however `tw-merge` is trivial to configure so this is not an issue in practice

> style deduplication is one of the most important features

not too sure about this. because of variants, we probably end up with much more generated CSS.

> We have `tailwind-variants`

not sure how this is different to (or better than) using `@apply` directly


also remember: the goal of tailwind is not to golf your css. the goal of tailwind is not to deduplicate styles. the goal of tailwind is to encode a design system, and arbitrary pixel values are not a design system.

MrFlashAccount replied:
> not too sure about this. because of variants, we probably end up with much more generated CSS.

It's not

> not sure how this is different to (or better than) using @apply directly

Apply is considered a bad practice by the authors of tw.

> you're not supposed to

Yes we supposed to. Because we. read. the. code.

MrFlashAccount replied:
We're not going to use `m-floating-buttons`, `button` or any other classes. Period

MrFlashAccount commented on app/ide-desktop/lib/dashboard/src/components/dashboard/ProjectIcon.tsx:
```diff
@@ -204,14 +184,35 @@ export default function ProjectIcon(props: ProjectIconProps) {
   )
 
   React.useEffect(() => {
```
Please don't do async logic in useEffect. Use `useQuery` or `useMutation` instead.

MrFlashAccount commented on app/ide-desktop/lib/dashboard/src/components/dashboard/ProjectIcon.tsx:
```diff
@@ -272,6 +266,15 @@ export default function ProjectIcon(props: ProjectIconProps) {
           setShouldSwitchPage(event.shouldAutomaticallySwitchPage)
           setIsRunningInBackground(event.runInBackground)
           void openProject(event.runInBackground)
+          void backend.getProjectDetails(item.id, item.parentId, item.title).then(project => {
```
Do not fetch directly in event handles, use `react-query`

MrFlashAccount commented on app/ide-desktop/lib/dashboard/src/components/dashboard/ProjectNameColumn.tsx:
```diff
@@ -73,10 +71,9 @@ export default function ProjectNameColumn(props: ProjectNameColumnProps) {
     (backend.type === backendModule.BackendType.local ||
       (ownPermission != null &&
         permissions.PERMISSION_ACTION_CAN_EXECUTE[ownPermission.permission]))
+  const isCloud = backend.type !== backendModule.BackendType.local
```
This check doesn't mean that we're using cloud

somebody1234 replied:
oh?

MrFlashAccount replied:
`backend.type !== backendModule.BackendType.local` means that we're using a backend that is not local.
Use `backend.type === backendModule.BackendType.remote` instead

MrFlashAccount commented on app/ide-desktop/lib/dashboard/src/components/styled/Button.tsx:
```diff
@@ -62,13 +65,13 @@ function Button(props: ButtonProps, ref: React.ForwardedRef<HTMLButtonElement>)
       >
         <div
-          className={`group flex selectable ${isDisabled || softDisabled ? 'disabled' : ''} ${active ? 'active' : ''}`}
+          className={`group flex selectable ${light ? 'selectable-light' : ''} ${isDisabled || softDisabled ? 'disabled' : ''} ${active ? 'active' : ''}`}
```
Do not build class list using template literals, use either `tailwind-merge` or `clsx`

MrFlashAccount commented on app/ide-desktop/lib/dashboard/src/components/styled/WindowButton.tsx:
```diff
@@ -0,0 +1,42 @@
+}
+
+/** A button for performing an action on a window. */
+export default function WindowButton(props: WindowButtonProps) {
```
We already have `CloseButton` that does the same and looks better

MrFlashAccount commented on app/ide-desktop/lib/dashboard/src/layouts/AssetsTable.tsx:
```diff
@@ -2469,7 +2489,26 @@ export default function AssetsTable(props: AssetsTableProps) {
+            })
+          }}
+        >
+          <UnstyledButton
```
Don't use `UnstyledButton`, use `ariaComponents.Button` with `custom` variant/size instead

MrFlashAccount commented on app/ide-desktop/lib/dashboard/src/layouts/Drive.tsx:
```diff
@@ -221,50 +212,60 @@ export default function Drive(props: DriveProps) {
-      setNewLabelNames(labelNames => new Set([...labelNames, newLabelName]))
-      setLabels(oldLabels => [...oldLabels, placeholderLabel])
-      try {
-        const newLabel = await backend.createTag({ value, color })
```
Don't do mutations in `useCallback`. Use `useMutation` instead

MrFlashAccount commented on app/ide-desktop/lib/dashboard/src/layouts/Drive.tsx:
```diff
@@ -221,50 +212,60 @@ export default function Drive(props: DriveProps) {
+            oldLabels.map(oldLabel => (oldLabel.id === placeholderLabel.id ? newLabel : oldLabel))
+          )
+        } catch (error) {
+          toastAndLog(null, error)
```
Do not toast errors.

MrFlashAccount commented on app/ide-desktop/lib/dashboard/src/layouts/Drive.tsx:
```diff
@@ -221,50 +212,60 @@ export default function Drive(props: DriveProps) {
+        setDeletedLabelNames(oldNames => new Set([...oldNames, value]))
+        setQuery(oldQuery => oldQuery.deleteFromEveryTerm({ labels: [value] }))
+        try {
+          await remoteBackend.deleteTag(id, value)
```
do not fetch outside of `useQuery` / `useMutation`

MrFlashAccount commented on app/ide-desktop/lib/dashboard/src/layouts/DriveBar.tsx:
```diff
@@ -112,28 +108,38 @@ export default function DriveBar(props: DriveBarProps) {
         <div className="flex h-row py-drive-bar-y">
           <HorizontalMenuBar>
+            <aria.DialogTrigger>
+              <UnstyledButton
```
Don't use `UnstyledButton`

MrFlashAccount commented on app/ide-desktop/lib/dashboard/src/layouts/Settings.tsx:
```diff
@@ -72,32 +72,37 @@ export default function Settings() {
   }, [sessionType, backend])
 
-  let content: React.JSX.Element
+  let content: JSX.Element | null
```
don't use `JSX.Element` since it's deprecated

MrFlashAccount commented on app/ide-desktop/lib/dashboard/src/layouts/Settings.tsx:
```diff
@@ -72,32 +72,37 @@ export default function Settings() {
   switch (settingsTab) {
     case SettingsTab.account: {
-      content = <AccountSettingsTab />
+      content = backend == null ? null : <AccountSettingsTab backend={backend} />
```
If backend is null then show loader. Don't pass null.

somebody1234 replied:
loader makes no sense here, if backend is null we cannot show the tab at all

MrFlashAccount replied:
When the backend could be null?

MrFlashAccount replied:
> loader makes no sense here, if backend is null we cannot show the tab at all

Do we exclude the tab from the sidebar then?

somebody1234 replied:
@MrFlashAccount currently yes (iirc). in the future we will probably want to show it under a paywall instead.

MrFlashAccount replied:
How's it related?

somebody1234 replied:
in the PR the backend is refactored out to [a context object](https://github.com/enso-org/enso/blob/377e23da4a29cebc907ac2abb244b73e82133615/app/ide-desktop/lib/dashboard/src/layouts/Settings/settingsData.tsx#L374)

somebody1234 replied:
how's what related?

MrFlashAccount replied:
I have no idea how backend == null is related to paywalls

MrFlashAccount replied:
And at this point I don't understand why the backend could be null in settings

somebody1234 replied:
@MrFlashAccount true... i was thinking that the tabs *could* be visible for the purpose of showing paywalls. but of course, if `backend == null` then the user is not logged in (so we don't know the user's plan, so no paywalls)

MrFlashAccount replied:
Is it possible now to show settings without logging in? Also, I think this case should be handled by the router(if a user is not logged in, redirect him to the login screen). WDYT?

somebody1234 replied:
> Is it possible now to show settings without logging in?

no idea. i guess *technically* no because it's no longer possible to avoid logging in?

> Also, I think this case should be handled by the router(if a user is not logged in, redirect him to the login screen). WDYT?

depends on whether we ever want to enable the app for users that are not logged in (e.g. custom builds without the cloud backend). not sure what happens with an *expired* session token when the user is physically offline, but we may need to take that into account too.

so tl;dr: *assuming we will never ever need to show settings without logging in* then it makes sense to `invariant` this. otherwise i don't think it hurts to retain this logic.

it's possible that we may be able to make this tidier by using a `Router` and `Layout`s here? thoughts?

MrFlashAccount replied:
> it's possible that we may be able to make this tidier by using a Router and Layouts here? thoughts?

![CleanShot 2024-05-29 at 12 40 24@2x](https://github.com/enso-org/enso/assets/61194245/fc58216b-5e6f-42d5-a5e6-ed0471a26862) 
![CleanShot 2024-05-29 at 12 41 06@2x](https://github.com/enso-org/enso/assets/61194245/04c5209e-29bb-47aa-b20b-e84be2203a75)

Looks like this is exactly how it's working right now.

MrFlashAccount replied:
> no idea. i guess technically no because it's no longer possible to avoid logging in?

Yeah, seems like it's not possible anymore

MrFlashAccount replied:
> depends on whether we ever want to enable the app for users that are not logged in (e.g. custom builds without the cloud backend). not sure what happens with an expired session token when the user is physically offline, but we may need to take that into account too.


Yeah, the behavior of the dashboard, when a user goes offline is still unclear... I think we need to revisit this part one day.

somebody1234 replied:
(re: routes)

right, but `remoteBackend` may still be `null` because the user might be in offline mode (`session.type === 'offline'`). not sure where it stands today (see above re: using the app offline inside e.g. company proxy)

so the `Layout`s here would be to distinguish between settings tab that need the backend and those that don't

MrFlashAccount commented on app/ide-desktop/lib/dashboard/src/layouts/Settings/DeleteUserAccountSettingsSection.tsx:
```diff
@@ -12,15 +11,24 @@ import UnstyledButton from '#/components/UnstyledButton'
 
+/** Props for a {@link DeleteUserAccountSettingsSection}. */
+export interface DeleteUserAccountSettingsSectionProps {
+  readonly backend: Backend
```
Why do we pass backend instead of consuming it from the `Context`? Props drilling is a bad practice.

somebody1234 replied:
this is not props drilling, it is one level deep. putting everything in a context to hide state is a bad practice

MrFlashAccount replied:
If we need backend deeper inside the React components tree, do we need to pass it as a prop or we can use context instead?

somebody1234 replied:
typically context (like how it is used everywhere else). i just figured that here specifically the root `Settings` component should be controlling basically all the state of its children

somebody1234 replied:
the settings search PR is probably vaguely related

MrFlashAccount replied:
So why do we pass the backend directly then?

MrFlashAccount replied:
Looks like a stylistic choice that makes no sense and leads to unneeded cognitive load and merge conflicts.
Like a change only to make a change.

somebody1234 replied:
> unneeded cognitive load 

imo contexts are a higher cognitive load since it is much harder to track where the state is coming from, let alone what its lifecycle looks like (in contrast, with props it's obvious where the state comes from - the parent). if you have a different opinion please do elaborate

> merge conflicts

explain

somebody1234 replied:
re: merge conflicts specifically - this is something that should never change, so i'm not sure whether this will be a problem (like at all)

also note that the main *reason* why this change is happening is so that the main `Settings` component controls all the state that is passed to each of its children. this means that the inputs are only ever acquired in one place, so that if the inputs change, only the hook in the main `Settings.tsx` needs to be changed, not every such occurrence of the hook in the entire `Settings/` subfolder using a hacky regex search+replace.

MrFlashAccount commented on app/ide-desktop/lib/dashboard/src/layouts/Settings/MembersSettingsTab.tsx:
```diff
@@ -15,94 +15,46 @@ import SettingsSection from '#/components/styled/settings/SettingsSection'
   return (
     <SettingsPage>
       <SettingsSection noFocusArea title={getText('members')} className="overflow-hidden">
-        <MembersSettingsTabBar />
```
Did you just remove the `invitations` ? Please revert the deletion

MrFlashAccount commented on app/ide-desktop/lib/dashboard/src/layouts/Settings/MembersTable.tsx:
```diff
@@ -32,25 +33,23 @@ export interface MembersTableProps {
-  )
+  const membersQuery = reactQuery.useQuery({
+    initialData: () => (populateWithSelf && user != null ? [user] : null),
+    queryKey: ['members'],
```
use `listUsers` key instead

MrFlashAccount commented on app/ide-desktop/lib/dashboard/src/layouts/Start.tsx:
```diff
@@ -0,0 +1,49 @@
+}
+
+/** Home screen. */
+export default function Home(props: HomeProps) {
```
If it's a dialog, why it's in layouts? Also the name doesn't represent tha this is a dialog component

MrFlashAccount commented on app/ide-desktop/lib/dashboard/src/providers/BackendProvider.tsx:
```diff
@@ -52,32 +35,74 @@ export interface BackendProviderProps extends Readonly<React.PropsWithChildren>
+  const remoteBackend = React.useContext(BackendContext).remoteBackend
+  if (remoteBackend == null) {
+    // eslint-disable-next-line no-restricted-syntax
+    throw new Error('This component requires a Cloud Backend to function.')
```
use `invariant` instead

MrFlashAccount commented on app/ide-desktop/lib/dashboard/src/providers/BackendProvider.tsx:
```diff
@@ -52,32 +35,74 @@ export interface BackendProviderProps extends Readonly<React.PropsWithChildren>
+  if (categoryModule.isCloud(category)) {
+    if (remoteBackend == null) {
+      // eslint-disable-next-line no-restricted-syntax
+      throw new Error(
```
use invariant instead

MrFlashAccount commented on app/ide-desktop/lib/dashboard/src/services/RemoteBackend.ts:
```diff
@@ -71,6 +71,7 @@ export async function waitUntilProjectIsReady(
     await remoteBackend.openProject(item.id, null, item.title)
   }
   let nextCheckTimestamp = 0
+  console.log(abortController.signal.aborted, project.state.type)
```
Remove please console.log

MrFlashAccount commented on app/ide-desktop/lib/dashboard/src/tailwind.css:
```diff
@@ -156,11 +156,12 @@
   --auth-link-padding-y: 0.25rem;
   --text-link-padding-x: 0.25rem;
+  /* The margin between a floating button and the edge of its window. */
+  --floating-buttons-margin: 1.1875rem;
```
Don't use custom margins

MrFlashAccount commented on app/ide-desktop/lib/dashboard/tailwind.config.js:
```diff
@@ -19,11 +19,14 @@ export default /** @satisfies {import('tailwindcss').Config} */ ({
         'ide-bg': '#ebeef1',
         selected: 'rgb(255 255 255 / 40%)',
+        'not-selected': 'rgb(0 0 0 / 15%)',
+        'window-close': '#ff605c',
```
use generic names instead of specific names

somebody1234 replied:
this is the generic name, this is the macos close color

MrFlashAccount commented on app/ide-desktop/lib/dashboard/tailwind.config.js:
```diff
@@ -94,6 +98,7 @@ export default /** @satisfies {import('tailwindcss').Config} */ ({
         'chat-profile-picture': 'var(--chat-profile-picture-size)',
         'selection-brush-border': 'var(--selection-brush-border-width)',
         'button-focus-ring-inset': 'var(--button-focus-ring-inset)',
+        'negative-button-focus-ring-inset': 'calc(var(--button-focus-ring-inset) * -1)',
```
don't use custom spacings

MrFlashAccount commented on app/ide-desktop/lib/dashboard/tailwind.config.js:
```diff
@@ -345,6 +346,7 @@ export default /** @satisfies {import('tailwindcss').Config} */ ({
         'date-input-calendar-gap': 'var(--date-input-calendar-gap)',
         'context-menu-macos-half-x': 'var(--context-menu-macos-half-width)',
         'context-menu-half-x': 'var(--context-menu-half-width)',
+        'floating-buttons': 'var(--floating-buttons-margin)',
```
use only generic classes

MrFlashAccount commented on app/ide-desktop/lib/dashboard/tailwind.config.js:
```diff
@@ -498,6 +501,9 @@ inset 0 -36px 51px -51px #00000014`,
               '',
           },
+          '.selectable-light': {
+            '@apply opacity-25': '',
```
don't use appy and custom classes unless you have to

MrFlashAccount commented:
QA ⛔️

1. Can we improve the drag experience to look similar to Google Drive? Currently, if user have a lot of assets, the DND button is hidden.
https://github.com/enso-org/enso/assets/61194245/066d2aa7-cbfe-415f-88ac-2e8b4738fa43
2. Invitations disappeared in settings
3. Buttons are not aligned (use `ariaComponents.Button`)
![CleanShot 2024-05-27 at 12 49 30](https://github.com/enso-org/enso/assets/61194245/138cd9f5-d8cb-40ba-807d-429634214663)

somebody1234 committed:
Address code review (part 1)

somebody1234 committed:
Address code review (part 2)

somebody1234 committed:
WIP: Replace template literal `className`s with `twMerge` or `tv`

somebody1234 commented:
re: 3 - most style issues were fixed in the second PR. i figured it's easier to implement it once rather than having to backport. thoughts?

somebody1234 committed:
Finishing replacing template literal `className`s with `twMerge` or `tv`

somebody1234 committed:
Fix docs

somebody1234 committed:
Remove `clsx`

somebody1234 committed:
Continue addressing code review

somebody1234 committed:
Merge branch 'develop' into wip/sb/dashboard-122

somebody1234 committed:
WIP: Switch to mutations for interacting with the backend

somebody1234 committed:
Initial implementation of proper React Queries for backend

somebody1234 committed:
Finish switching to React Query mutations

somebody1234 committed:
Continue migration to React Query mutations

somebody1234 committed:
Replace `UnstyledButton` with `ariaComponents.Button`

somebody1234 committed:
Minor style fixes

somebody1234 commented:
2 & 3. should be addressed.
1. seems like it's a bigger feature, but *hopefully* it's the last thing to address (after some other final refactors)

somebody1234 committed:
Key backend queries and mutations on the actual backend object itself

somebody1234 committed:
Replace `setOrganization` in settings with React Query mutations

somebody1234 committed:
Fix bugs related to updating organizations and users

somebody1234 committed:
Merge branch 'develop' into wip/sb/dashboard-122

somebody1234 committed:
Fix type errors and lint errors

somebody1234 commented:
on second thought 1 will be left for later

somebody1234 committed:
Fix dropzone styling

somebody1234 committed:
Merge branch 'develop' into wip/sb/dashboard-122

somebody1234 committed:
Add dropzone

somebody1234 committed:
Fix chat CSS

somebody1234 committed:
Switch context menu away from Button

somebody1234 committed:
Fix drag-n-drop bug

somebody1234 committed:
Fix E2E tests

somebody1234 committed:
WIP: Fix opening projects

somebody1234 committed:
Fix React Query mutation effect dependencies

somebody1234 committed:
Fix `ProjectIcon` appearance

somebody1234 committed:
Add mutation listeners for labels

somebody1234 committed:
Merge branch 'develop' into wip/sb/dashboard-122

somebody1234 committed:
Merge branch 'develop' into wip/sb/dashboard-122

somebody1234 committed:
Fix E2E tests

somebody1234 committed:
Fix lint errors

PabloBuchu commented:
1. After clicking `Start with a template` and closing modal I end up with this
<img width="316" alt="Screenshot 2024-06-10 at 13 40 47" src="https://github.com/enso-org/enso/assets/2855109/ec93c9be-23b3-40e7-bc1d-42f1dcb4d56c">

2. `New empty project` and polling for status is bugged
<img width="1049" alt="Screenshot 2024-06-10 at 13 41 18" src="https://github.com/enso-org/enso/assets/2855109/9ee25f80-d751-4f80-a9ef-fe7e6dfd8928">
<img width="1709" alt="Screenshot 2024-06-10 at 13 41 12" src="https://github.com/enso-org/enso/assets/2855109/5b5dd4a7-366d-497e-adaa-a71e3476eeca">

3. Should this be so weird
<img width="113" alt="Screenshot 2024-06-10 at 13 43 06" src="https://github.com/enso-org/enso/assets/2855109/83074421-4680-4082-b1f5-85eb8d6143c5">

4. Tooltips problem
https://github.com/enso-org/enso/assets/2855109/db6802ae-ded6-4b8d-b7ec-f5a96985ae9c

5. Show secret value is not aligned and doesn't work
<img width="402" alt="Screenshot 2024-06-10 at 13 44 56" src="https://github.com/enso-org/enso/assets/2855109/9c160318-63ae-40d7-aacd-4d691804f7af">

6. Share with modal
<img width="453" alt="Screenshot 2024-06-10 at 13 46 05" src="https://github.com/enso-org/enso/assets/2855109/a90b7b38-e24e-4e31-9ad8-2dcc53a6aaad">

7. Have we disabled cmd / shift + arrows multiple selection?

somebody1234 commented:
uhoh

somebody1234 committed:
Address QA

somebody1234 commented:
fixed some things locally, some of them are still TODO:
1. fixed, assuming the issue is the focus ring being cut off
  - i've noticed the focus ring is inconsistent (spacing, animations) - this should probably be fixed at some point as well
2. TODO
3. fixed
4. fixed, assuming the issue is the positioning causing issues
5. fixed
6. fixed
7. (debugging TODO) idts - but it seems to be bugged, i can see the selection flashing for a moment before being reset to just the one row

PabloBuchu commented:
@somebody1234 we can fix 7th on separate PR and then just fix 2nd issue here. Then we can push this to merge after all others are fixed

somebody1234 commented:
found the cause of 2nd, was me being silly. @PabloBuchu should be ready for re-QA now - although now's probably not the best time, lol

somebody1234 commented:
@MrFlashAccount just a note that this PR is blocked on your re-review

somebody1234 committed:
Prettier

PabloBuchu commented:
QA passed. I created one small fix https://github.com/enso-org/cloud-v2/issues/1301

lets fix the merge conflict and try to merge it today

@MrFlashAccount can you rereview and approve?

somebody1234 committed:
Merge branch 'develop' into wip/sb/dashboard-122
